experiments/download_ssl4eo.py

```diff
@@ -396,23 +393,20 @@ def get_random_patches_grid(
     # avoid strong overlap
     try:
         new_coord = (coords[0], coords[1])
-        gridIndex = (math.floor(new_coord[0] + 180), math.floor(new_coord[1] + 90))
-
-        if gridIndex not in grid_dict.keys():
-            grid_dict[gridIndex] = {new_coord}
-        else:
-            for coord in grid_dict[gridIndex]:
-                distance = np.sqrt(
-                    sampler.deg2km(abs(new_coord[0] - coord[0])) ** 2
-                    + sampler.deg2km(abs(new_coord[1] - coord[1])) ** 2
-                )
-                if distance < (1.5 * radius / 1000):
-                    raise OverlapError
-            grid_dict[gridIndex].add(new_coord)
+        for i in rtree_obj.nearest(new_coord, num_results=1, objects=True):
+            distance = np.sqrt(
+                sampler.deg2km(abs(new_coord[0] - i.bbox[2])) ** 2
+                + sampler.deg2km(abs(new_coord[1] - i.bbox[3])) ** 2
+            )
+            if distance < (1.5 * radius / 1000):
+                raise OverlapError
+        rtree_obj.insert(
+            len(rtree_obj) - 1, (new_coord[0], new_coord[1], new_coord[0], new_coord[1])
+        )
 
     except OverlapError:
-        patches_s1, patches_s2c, patches_s2a, center_coord = get_random_patches_grid(
-            idx, collections, bands, crops, sampler, dates, radius, debug
+        patches_s1, patches_s2c, patches_s2a, center_coord = get_random_patches_rtree(
```

Comment:
Is recursion necessary here? It could possibly end up with a very deep recursion stack. I would personally use a `while True` loop and break out when you find a valid location. Then you don't need to raise and catch errors, should be much simpler.

Reply:
No, didn't think much about the difference. I thought recursion and loop work the same here?

Reply:
In some compiled languages like OCaml where tail recursion is automatically converted to a while loop, yes. But in Python, recursion has a lot more overhead than a while loop since you need to store all inputs at every step of the recursion stack.

Reply:
i see, i've updated with while loop